modules/dbnd-airflow/src/dbnd_airflow/scheduler/single_dag_run_job.py

```diff
@@ -721,14 +720,27 @@ def _execute(self, session=None):
                     ti_status.executed_dag_run_dates,
                 )
         finally:
-            # in sequential executor a keyboard interrupt would reach here and then executor.end() -> heartbeat() -> sync() will cause the queued commands
+            # in sequential executor a keyboard interrupt would reach here and
+            # then executor.end() -> heartbeat() -> sync() will cause the queued commands
             # to be run again before exiting
             if hasattr(executor, "commands_to_run"):
                 executor.commands_to_run = []
             try:
                 executor.end()
             except Exception:
                 logger.exception("Failed to terminate executor")
+            if dag_run.state not in (State.SUCCESS, State.FAILED):
+                TI = TaskInstance
+                session.query(TI).filter(
+                    TI.dag_id == self.dag_id,
+                    TI.execution_date == self.execution_date,
+                    TI.state != State.SUCCESS,
+                ).update(
```

Comment:
I would separate query here. we run it only once in a while, we better have a good debug info on what we fix